[ews-build] macOS WK2 Intel tests and arm64 tests share generated S3 URLs

briannafan opened the pull request:
#### 483f8e31a7ad3e40cb3d9eeab99659ebf5464a05
<pre>
[ews-build] macOS WK2 Intel tests and arm64 tests share generated S3 URLs
<a href="https://bugs.webkit.org/show_bug.cgi?id=284333">https://bugs.webkit.org/show_bug.cgi?id=284333</a>
<a href="https://rdar.apple.com/141183934">rdar://141183934</a>

Reviewed by Ryan Haddad.

* Tools/CISupport/Shared/generate-s3-url:
(generateS3URL): Add additional suffix to file keys if specified.
(main): Add &apos;additions&apos; option.

* Tools/CISupport/ews-build/steps.py: Add `additions=f&apos;self.build.number&apos;`
                                      to all .txt invocations of GenerateS3URL.
(CompileWebKit.follow_up_steps):
(RunJavaScriptCoreTests.evaluateCommand):
(RunWebKitTests.evaluateCommand):
(RunWebKitTestsInStressMode.evaluateCommand):
(ReRunWebKitTests.evaluateCommand):
(RunWebKitTestsWithoutChange.evaluateCommand):
(RunWebKitTestsRedTree.evaluateCommand):
(RunWebKitTestsRepeatFailuresRedTree.evaluateCommand):
(RunWebKitTestsRepeatFailuresWithoutChangeRedTree.evaluateCommand):
(RunWebKitTestsWithoutChangeRedTree.evaluateCommand):
(RunAPITests.run):

(GenerateS3URL.__init__): Add `additions` argument.
(GenerateS3URL.run):
(GenerateS3URL):
(GenerateS3URL.hideStepIf): Deleted.

* Tools/CISupport/ews-build/steps_unittest.py:
(TestGenerateS3URL.configureStep):
(TestGenerateS3URL.test_failure):

Canonical link: <a href="https://commits.webkit.org/288170@main">https://commits.webkit.org/288170@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2d6fc8149cdc87cb56b356ce8f33caac2f0131e1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/82064 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/1591 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/36021 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/86621 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/33096 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/84170 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/1626 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/9418 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/86621 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/33096 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/85134 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/1209 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/74664 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/86621 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/1113 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/28842 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/31516 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/72423 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/29453 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/88054 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/9306 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/6649 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/72342 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/81701 "Passed tests") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/9491 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/70483 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/71564 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/126/builds/15692 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/14604 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/721 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/12718 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/9257 "Built successfully") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/9097 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/12623 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/10905 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

ryanhaddad commented on Tools/CISupport/ews-build/config.json:
```diff
@@ -285,7 +285,7 @@
       "name": "macOS-Ventura-Release-WK2-Intel-Tests-EWS", "shortname": "mac-intel-wk2", "icon": "testOnly",
       "factory": "macOSWK2Factory", "platform": "mac-ventura",
-      "configuration": "release",
+      "configuration": "release", "architectures": ["x86_64"],
```
I'm concerned that this will cause the `download-built-product` step to fail. The [architectures are present in those paths as well](https://ews-build.webkit.org/#/builders/8/builds/68819/steps/14/logs/stdio), and the triggering queues are indeed building both architectures.

ryanhaddad reviewed (changes requested):
Marking as "Request changes" for now, since we need to think about this some more.

ryanhaddad commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -5328,14 +5341,15 @@ def run(self):
-        return results == SUCCESS
+    # TODO: Remove after testing. Also worth thinking about whether we want to hide this or not.
+    #def hideStepIf(self, results, step):
+    #    return results == SUCCESS
```
TODO

briannafan committed:
[ews-build] macOS WK2 Intel tests and arm64 tests share generated S3 URLs
https://bugs.webkit.org/show_bug.cgi?id=284333
rdar://141183934

Reviewed by Ryan Haddad.

* Tools/CISupport/Shared/generate-s3-url:
(generateS3URL): Add additional suffix to file keys if specified.
(main): Add 'additions' option.

* Tools/CISupport/ews-build/steps.py: Add `additions=f'self.build.number'`
                                      to all .txt invocations of GenerateS3URL.
(CompileWebKit.follow_up_steps):
(RunJavaScriptCoreTests.evaluateCommand):
(RunWebKitTests.evaluateCommand):
(RunWebKitTestsInStressMode.evaluateCommand):
(ReRunWebKitTests.evaluateCommand):
(RunWebKitTestsWithoutChange.evaluateCommand):
(RunWebKitTestsRedTree.evaluateCommand):
(RunWebKitTestsRepeatFailuresRedTree.evaluateCommand):
(RunWebKitTestsRepeatFailuresWithoutChangeRedTree.evaluateCommand):
(RunWebKitTestsWithoutChangeRedTree.evaluateCommand):
(RunAPITests.run):

(GenerateS3URL.__init__): Add `additions` argument.
(GenerateS3URL.run):
(GenerateS3URL):
(GenerateS3URL.hideStepIf): Deleted.

* Tools/CISupport/ews-build/steps_unittest.py:
(TestGenerateS3URL.configureStep):
(TestGenerateS3URL.test_failure):

Canonical link: https://commits.webkit.org/288170@main
